src/ConnectedMode/UI/BindingControllerAdapter.cs

```diff
@@ -0,0 +1,115 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.ConnectedMode.Binding;
+using SonarLint.VisualStudio.ConnectedMode.Shared;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.SLCore.Protocol;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UI;
+
+internal interface IBindingControllerAdapter
+{
+    Task<BindingResult> ValidateAndBindAsync(BindingRequest request, IConnectedModeUIManager uiManager, CancellationToken token);
+
+    bool Unbind(string bindingKey = null);
+}
+
+[Export(typeof(IBindingControllerAdapter))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+[method: ImportingConstructor]
+internal class BindingControllerAdapter(
+    IBindingController bindingController,
+    ISolutionInfoProvider solutionInfoProvider,
+    IServerConnectionsRepositoryAdapter serverConnectionsRepositoryAdapter,
+    ILogger logger)
+    : IBindingControllerAdapter
+{
+    private readonly ILogger logger = logger.ForContext(ConnectedMode.Resources.ConnectedModeLogContext, ConnectedMode.Resources.ConnectedModeBindingLogContext);
+
+    public async Task<BindingResult> ValidateAndBindAsync(BindingRequest request, IConnectedModeUIManager uiManager, CancellationToken token)
```

Comment:
New adapter which has the functionality that was previously in ManageBindingViewModel